modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -1765,6 +1760,21 @@ private CompletableFuture<Void> applyWriteIntentSwitchCommand(
                 .thenApply(res -> null);
     }
 
+    private void switchWriteIntentsUnderSnapshotsLock(UUID transactionId, boolean commit, HybridTimestamp commitTimestamp) {
+        storageUpdateHandler.acquirePartitionSnapshotsReadLock();
+
+        try {
+            storageUpdateHandler.switchWriteIntents(
+                    transactionId,
+                    commit,
+                    commitTimestamp,
+                    indexIdsAtRwTxBeginTs(transactionId)
+            );
+        } finally {
+            storageUpdateHandler.releasePartitionSnapshotsReadLock();
```

Comment:
I'm not sure that we should use snapshot lock within partition replica listener. I've scheduled a meeting to discuss this.